File: bench/ruby/pipeline.rb
n = (ENV["BENCH_N"] || "200000").to_i

# map / filter / reduce pipeline: square the multiples of 3 or 5, sum the squares.
total = (0...n).select { |i| i % 3 == 0 || i % 5 == 0 }.map { |i| i * i }.reduce(0, :+)
puts total

File: bench/python/pipeline.py
import os
N = int(os.environ.get("BENCH_N", "200000"))

# map / filter / reduce pipeline: square the multiples of 3 or 5, sum the squares.
total = sum(i * i for i in range(N) if i % 3 == 0 or i % 5 == 0)
print(total)
